core/package/index.js

```diff
@@ -185,6 +185,11 @@ class Package {
       depCollection = this.devDependencies;
     }
 
+    // fall back to peerDependencies
+    if (!depCollection || !depCollection[depName]) {
+      depCollection = this.peerDependencies;
```

Comment:
As @ljharb [noted](https://github.com/lerna/lerna/issues/955#issuecomment-332074663) in the linked issue, unconditionally updating the minimum range of peer dependencies like this would constitute a breaking change. You'd need a completely separate branch of logic to properly update it, IFF `semver.satisfies(depVersion, peerDeps[depName])` returns `false`, and _then_ you'd need to preserve whatever the current value was (if any) and either append a new alternation (`|| ^{depVersion}`, e.g., for a new major) or set a new value (I'm not even sure that's valid, since _adding_ new peers should _always_ be explicit).